core/domain/exp_services.py

```diff
@@ -29,6 +29,7 @@
 import pprint
 import StringIO
 import zipfile
+from collections import defaultdict
```

Comment:
Do "import collections" (at the top; imports are sorted alphabetically) and refer to collections.defaultdict in the file